### ISSUE TITLE
Reproducing MLP performance on PascalVOC

### ISSUE THREAD

**Issue by acharkq** [rusty1s/deep-graph-matching-consensus#17]
Hi, I am interested in training a weak feature matching model and want to reproduce the MLP's performance in Table1.

Specifically, I want to reproduce the performance without using DGMC, the row of MLP isotropic L=0.

I have reused the feature loading and training code in `pascal.py`. However, my performance is only `43.99`.

The following is my MLP implementation. Would you give some advice on improving the performance? Thanks!

```python
class MLP(nn.Module):
    def __init__(self, in_dim, out_dim, hid_dim, dropout=0.5, num_layer=3):
        super(MLP, self).__init__()
        assert num_layer >= 1
        if num_layer == 1:
            hid_dim = out_dim

        self.layers = nn.ModuleList([])
        self.layers.append(nn.Dropout(dropout))
        self.layers.append(nn.Linear(in_dim, hid_dim))
        for i in range(num_layer - 1):
            self.layers.append(nn.ReLU())
            self.layers.append(nn.Dropout(dropout))
            if i == num_layer - 2:
                self.layers.append(nn.Linear(hid_dim, out_dim))
            else:
                self.layers.append(nn.Linear(hid_dim, hid_dim))
        self.residual = nn.Linear(in_dim, out_dim)

    def forward(self, x_s, x_t, batch_s, batch_t):
        h_s = x_s
        h_t = x_t
        for layer in self.layers:
            h_s = layer(h_s)
            h_t = layer(h_t)
        h_s = h_s + self.residual(x_s)
        h_t = h_t + self.residual(x_t)

        h_s, s_mask = to_dense_batch(h_s, batch_s, fill_value=0)
        h_t, t_mask = to_dense_batch(h_t, batch_t, fill_value=0)
        (B, N_s, C_out), N_t = h_s.size(), h_t.size(1)
        S_hat = h_s @ h_t.transpose(-1, -2)  # [B, N_s, N_t, C_out]
        S_mask = s_mask.view(B, N_s, 1) & t_mask.view(B, 1, N_t)
        S = masked_softmax(S_hat, S_mask, dim=-1)[s_mask]
        return S
```

**Comment by acharkq** [comment]
As an update, I have tried your MLP implementation in `models.mlp.py`. I have trained the model for 100 epochs and the best performance on test set is `49.3`.

I am now setup a training of 1000 epochs and want to select the best model based on the test performance.

```python
MLP(dataset.num_node_features, 256, 3, batch_norm=True, dropout=0.5)
optimizer = torch.optim.Adam(model.parameters(), lr=0.001)
```

May I know your choice of hyperparameters?

**Comment by rusty1s** [comment]
I sadly didn't have saved the exact MLP configuration :( I remember that I did a quick hyperparameter sweep. In particular, `batch_norm = { True, False }`, `num_layers = {2, 3}`, and `dropout = {0.0, 0.5}`.
